portfolio/src/main/java/com/google/sps/PuzzleDao.java

```diff
@@ -59,10 +61,20 @@ public Puzzle read(Long puzzleId) {
 
       //saves the retrieved data into a "Puzzle" object
       Difficulty difficulty = Puzzle.Difficulty.valueOf(entity.getString("difficulty"));
+
+      String name;
+
+      try {
+          name = entity.getString("name");
+      } catch (Exception e) {
```

Comment:
It's generally discouraged to catch Exception because it's very expansive. Always prefer to catch as specific of an exception as possible. In this case I think it is `DatastoreException` for cases where the property doesn't exist([documentation](https://googleapis.dev/java/google-cloud-datastore/latest/index.html)):
```suggestion
      } catch (DatastoreException e) {
```